### ISSUE TITLE
Performance issue when downloading packages with versions after 1.3 and newer

### ISSUE THREAD

**Issue by wende** [gdotdesign/elm-github-install#42]
Before:
<img width="857" alt="screen shot 2017-09-05 at 17 45 11" src="https://user-images.githubusercontent.com/4634386/30069904-375dae20-9262-11e7-8d64-ea1c9ce5bea1.png">

On
<img width="885" alt="screen shot 2017-09-05 at 17 47 00" src="https://user-images.githubusercontent.com/4634386/30069921-42b21914-9262-11e7-975d-f95718d4240c.png">
 version 1.3+

**Comment by wende** [comment]
Same behaviour after cleaning the cache:
<img width="864" alt="screen shot 2017-09-05 at 17 49 15" src="https://user-images.githubusercontent.com/4634386/30070044-8ea4501c-9262-11e7-860a-c860752665e9.png">

**Comment by gdotdesign** [comment]
I'm not sure what is causing this, I'll try to debug it soon.

**Comment by gdotdesign** [comment]
Managed to track down the issue: during the resolving process we check each version against the elm-version of the current `elm-package.json`, this wasn't optimized and coupled with the high number of versions `elchemy-core` has lead to the increased runtime.

After the fix here are my numbers with the `elm-package.json`:
```json
{
  "dependencies": {
    "wende/elchemy-core": "0.0.0 < v <= 1.0.0"
  },
  "elm-version": "0.18.0 <= v < 0.19.0"
}
```
![after-fix](https://user-images.githubusercontent.com/302692/30474563-631e847c-9a04-11e7-9871-36815cc97905.png)

If you limit the version number then the runtime is much faster:
```json
{
  "dependencies": {
    "wende/elchemy-core": "0.4.36 < v <= 1.0.0"
  },
  "elm-version": "0.18.0 <= v < 0.19.0"
}
```
![after-fix-limited-version](https://user-images.githubusercontent.com/302692/30474622-973bf6ae-9a04-11e7-9037-cfd3a3b8df0c.png)

**Comment by gdotdesign** [comment]
Released as v1.5.0

**Comment by wende** [comment]
Beautiful! Thank you 🎉
